product-mini/platforms/posix/main.c

```diff
@@ -803,8 +811,31 @@ main(int argc, char *argv[])
                    patch);
             return 0;
         }
+
+        else if (!strncmp(argv[0], "--strace=", 9)) { // strace w/ log file
+            if (argv[0][9] == '\0')
+                return print_help();
+            char *the_file = argv[0] + 9;
+            strace = 1;
+            strace_logfile = fopen(the_file, "wb"); // test
+
+            if(strace_logfile == NULL)
+            {
+                printf(strerror(errno));
+                fflush(stdout);
+            }
+
+            atexit(close_log_file);
+            printf("strace file: %s.\n", the_file);
+        }
+
+        else if (!strncmp(argv[0], "--strace", 8)) { // strace print to console
+            strace = 0;
+            strace_logfile = stdout; // STDOUT
+        }
+        
```

Comment:
* You should remove the printf for the strace_file here: instead put it in wali_init as a verbose message